File: algorithms/Largest Divisible Subset/Largest Divisible Subset.cpp
class Solution {
public:
    vector<int> largestDivisibleSubset(vector<int>& nums) {
        
        int n = nums.size();
        sort(nums.begin(), nums.end());
        
        vector<int> ans;        
        for(int i = 0; i < n; i++)
        {                
            vector<int> tmp;
            tmp.push_back(nums[i]);
            
            // Checking the prev one
            int j = i - 1;
            while(j >= 0)
            {
                if(tmp.back() % nums[j] == 0)
                    tmp.push_back(nums[j]);
                    
                j--;
            }
            
            // Reversing the order to make it in increasing order
            reverse(tmp.begin(), tmp.end());
            
            // Checking the forward one
            j = i + 1;
            while(j < n)
            {
                if(nums[j] % tmp.back() == 0)
                    tmp.push_back(nums[j]);
                
                j++;
            }
            
            // updating the ans
            if(ans.size() < tmp.size())
                ans = tmp;
        }
        
        return ans;        
    }
};

File: algorithms/L/Largest Divisible Subset/Largest Divisible Subset.py
class Solution:
    def largestDivisibleSubset(self, nums: List[int]) -> List[int]:
        nums.sort()
        n = len(nums)
        dp = [1 for i in range(n)]
        hashh = [i for i in range(n)]
        ans_ind = 0
        
        for i in range(1, n):
            for j in range(0,i):
                if nums[i]%nums[j] == 0 and dp[j]+1 > dp[i]:            
                    dp[i] = dp[j]+1
                    hashh[i] = j
                    
                    # print(dp)
                    # print(hashh)
        out = []
        maxi = dp[0]
        
        for i in range(len(nums)):
            if dp[i] > maxi:
                ans_ind = i
                maxi = dp[i]
        
        while(hashh[ans_ind]!=ans_ind):
            out.append(nums[ans_ind])
            ans_ind = hashh[ans_ind]
        out.append(nums[ans_ind])
        return(out)
        
        
        
